lib/Frontend/CompilerInvocation.cpp

```diff
@@ -961,7 +961,7 @@ static bool ParseDiagnosticArgs(DiagnosticOptions &Opts, ArgList &Args,
     // Check if the locale code is available.
     if (llvm::none_of(localeCodes, [&](const char *locale) {
           return localeCode == locale;
-        })) {
+        }) && localeCode != "test") {
```

Comment:
Just a question. Why are we checking if `localeCode` doesn't equal to "test"?

Reply:
This is an unfortunate, but temporary hack to allow testing with `-locale test` since there aren't any real translations yet. `EducationalNotes.def` registers one note as having a translation for the "test" language so the tests can verify the changes work properly without adding a real translation. Once there's at least one, we can update the tests to use that instead and remove this.

Reply:
Thank you for the clarification. Do you think it's worth writing a comment about this?

Reply:
good idea, added